### ISSUE TITLE
DTLSv1_listen unable to accept second client in a docker container

### ISSUE THREAD

**Issue by alfonsoveneziano** [openssl/openssl#10019]
Environment: docker container based on CentOs7
OpenSSL version: OpenSSL-1.1.1d

A DTLS server (non-blocking) using DTLSv1_Listen having a UDP socket with SO_REUSEADDR is unable to accept a second 
client connection when it has already been accepted a client connection and serving it.
When the first client has finished, the second client connection is accepted.

We have used the dtls_udp_echo.c (taken from http://web.archive.org/web/20150617012520/http://sctp.fh-muenster.de/dtls-samples.html ) to carry out our tests.

The test application has been compiled and executed within a docker container, having CentOS7 as base image, but the behaviour has been noticed with other base images OS too (e.g. Redhat, Ubuntu, Debian, SLES).

The same application running on a bare metal works without any issue.

Attached you can find a zip file containing the Dockerfile to create docker images with :
1. **Readme** file with instructions to create the docker image, create the network and **to reproduce the issue**
2. OpenSSL library already built
3. dtls_udp_echo already built and private keys and certificates for the application

Is there any known compatibility issue between Docker and OpenSSL/DTLS?
Is there any specific configuration to be done to overcome this issue?

Best Regards

[openssl_issue.zip](https://github.com/openssl/openssl/files/3653759/openssl_issue.zip)

**Comment by mattcaswell** [comment]
> Is there any known compatibility issue between Docker and OpenSSL/DTLS?

There aren't any *known* compatibility issues. That doesn't rule out unknown ones. :-)

> Is there any specific configuration to be done to overcome this issue?

Not in OpenSSL AFAIK but quite possibly in Docker. If the same application is running fine on bare metal then it tends to suggest a docker configuration problem. Unfortunately my docker knowledge is non-existent so I'm not able to advise.

**Comment by alfonsoveneziano** [comment]
Hi
we have repeated a similat test just using plain socket connections and we have not seen any issue.
So we suspect that the problem could be somehow related to BIO, we don't have any evidence.
Is there any debugging tools (or similar) that we can use to further investigate and detect where the problem is?

Would be possible for you to reproduce the issue and exclude any relation with OpenSSL?

In the meanwhile we have submitted similar request to docker community. 

Unfortunately, if we don't find any solution in the short time we have to look for another library.

Thanks in advance
Best Regards
